test/unit-test/core_mqtt_utest.c

```diff
@@ -4545,6 +4915,63 @@ void test_MQTT_unsubscribe_happy_path2( void )
     TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
 }
 
+void test_MQTT_Unsubscribe_MultipleSubscriptions( void )
+{
+    MQTTStatus_t mqttStatus;
+    MQTTContext_t context = { 0 };
+    TransportInterface_t transport = { 0 };
+    MQTTFixedBuffer_t networkBuffer = { 0 };
+    MQTTSubscribeInfo_t subscribeInfo[ 5 ];
+    size_t remainingLength = MQTT_SAMPLE_REMAINING_LENGTH;
+    size_t packetSize = MQTT_SAMPLE_REMAINING_LENGTH;
+    MQTTPubAckInfo_t incomingRecords = { 0 };
+    MQTTPubAckInfo_t outgoingRecords = { 0 };
+
+    TEST_ASSERT_EQUAL_MESSAGE( 6U, MQTT_SUB_UNSUB_MAX_VECTORS,
+                               "This test is configured to work with MQTT_SUB_UNSUB_MAX_VECTORS defined as 6." );
+
+    setupTransportInterface( &transport );
+    transport.writev = transportWritevUnsubscribeSuccess;
+
+    setupNetworkBuffer( &networkBuffer );
+
+    subscribeInfo[ 0 ].qos = MQTTQoS1;
```

Comment:
Hmm, it can be. But I don't see much benefit since this unit test is written for one configuration. And we assert on the configuration value being correct.

If it were an integration test, then I would be more inclined to make this change.